### DISCUSSION TITLE
How does the mapping work for the AI inline suggestion?

### DISCUSSION THREAD

**Discussion by MatejBransky** [olimorris/codecompanion.nvim#318] [Q&A] [answered]
Hi,

I’m trying to figure out how to accept the inline suggestion offered by AI. I don’t use the Tab key for acceptance in nvim-cmp, and my cmp confirmation with `<C-y>` isn’t working in this case.

<img width="398" alt="image" src="https://github.com/user-attachments/assets/9e463d9e-2a93-400c-aad9-b2e84a541b72">

How can I set up a mapping for confirming AI suggestions?

I’d also like to know if it’s possible to accept only individual words because sometimes the suggested beginning is good, but I would like to write a bit myself so that the next suggestion is more likely to be what I need to add. Can that be configured as well?

In VS Code, partial acceptance of suggestions was handled with `<Cmd-RightArrow>`, I believe (see the attached recording).

https://github.com/user-attachments/assets/fef2d4d1-2c5f-4bf7-8391-5ce0d9504ee8

**Comment by olimorris** [comment, marked as answer]
The plugin doesn't make any suggestions unless you engage the Inline Assistant. In which case, `ga` or `gr` will accept/reject them.

**Reply by PLAZMAMA** [reply]
If the inline assistant current engages via a command, Is there a way then to create a custom event that would trigger the inline assistant when you're in insert mode? I will ideally love to have something that would work like how Copilot or Codeium does?

**Reply by MatejBransky** [reply]
Thank you for your response. 

Thanks to your answer, I found out that Copilot.vim is directly responsible for the inline suggestions, so I just had to adjust the [keyboard shortcuts in its configuration](https://github.com/github/copilot.vim/blob/release/doc/copilot.txt#L111).
